### PR TITLE
Jgfouca/branch for acme split 2018 01 09

### PR DESCRIPTION
**Pull request by jgfouca**
Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: N

Update gh-pages html (Y/N)?: N

Code review: @jedwards4b

### CODE REVIEW

**Comment by jgfouca** [comment]
Tests still running.

@agsalin @mt5555 @rljacob , it appears the integration is working again.

File: scripts/lib/acme_cime_mgmt.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def acme_cime_merge(resume):
         pr_branch = get_branch_from_tag(new_merge_tag)
 
     try:
-        run_cmd_no_fail("git push -u origin {}".format(pr_branch), verbose=True)
+        run_cmd_no_fail("git push -u origin {}".format(pr_branch))
```
Comment (jgfouca) [inline scripts/lib/acme_cime_mgmt.py]: This is not right.

File: src/drivers/mct/main/prep_ocn_mod.F90 [hunk 2]
```diff
@@ -48,6 +48,8 @@ module prep_ocn_mod
 
   public :: prep_ocn_get_x2oacc_ox
   public :: prep_ocn_get_x2oacc_ox_cnt
+  ! Sarat: Dummy variable added to workaround PGI compiler bug (PGI 17.9) as of Oct 23, 2017
+  public :: dummy_pgibugfix
```
Comment (jedwards4b) [inline src/drivers/mct/main/prep_ocn_mod.F90]: This is ugly - can we add 
#ifdef CPRPGI around it?

**Comment by jgfouca** [comment]
These changes were mostly exactly what I expected, the vast majority being xml changes to config/acme.

File: src/drivers/mct/main/prep_ocn_mod.F90 (hunk 2 above)
Comment (jgfouca) [reply src/drivers/mct/main/prep_ocn_mod.F90]: @amametjanov , can you comment here?

Comment (amametjanov) [reply src/drivers/mct/main/prep_ocn_mod.F90]: Looks like it's more specific than CPRPGI, was needed for v17.9  on the SummitDev machine: see https://github.com/ACME-Climate/ACME/pull/1908.

Comment (jedwards4b) [reply src/drivers/mct/main/prep_ocn_mod.F90]: We can at least limit it to PGI builds with that cppdef.

Comment (amametjanov) [reply src/drivers/mct/main/prep_ocn_mod.F90]: There is a new `SUMMITDEV_PGI` macro in that PR, so `#ifdef SUMMITDEV_PGI` should be just the right scope.

File: src/drivers/mct/main/prep_ocn_mod.F90 [hunk 3]
```diff
@@ -48,6 +48,10 @@ module prep_ocn_mod
 
   public :: prep_ocn_get_x2oacc_ox
   public :: prep_ocn_get_x2oacc_ox_cnt
+#ifdef SUMMITDEV_PGI
```
Comment (jgfouca) [inline src/drivers/mct/main/prep_ocn_mod.F90]: OK, I added the preprocessor restriction.

Comment (amametjanov) [reply src/drivers/mct/main/prep_ocn_mod.F90]: Also need to add this to references in `cime/src/drivers/mct/main/seq_rest_mod.F90`.

Comment (jgfouca) [reply src/drivers/mct/main/prep_ocn_mod.F90]: @amametjanov I don't see any references to "dummy" or "pgi" in that file.

Comment (amametjanov) [reply src/drivers/mct/main/prep_ocn_mod.F90]: Oh, the latest cime merge took out those `dummy_pgibugfix = associated(x2oacc_ox)` workarounds from that PR. @sarats may want to re-check with the latest E3SM master.

Comment (jgfouca) [reply src/drivers/mct/main/prep_ocn_mod.F90]: @amametjanov you're right, it was lost in the recent ESMCI integration, I will put it back.

Comment (jgfouca) [reply src/drivers/mct/main/prep_ocn_mod.F90]: @amametjanov @sarats , to avoid this in the future, it might be best to make driver changes under ESMCI.

Comment (sarats) [reply src/drivers/mct/main/prep_ocn_mod.F90]: @jgfouca Understood, thanks for fixing.

**Pull request merged**